src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2562,6 +2562,52 @@ def create_poly1305_ctx(self, key):
 
         return _Poly1305Context(self, key)
 
+    def load_pem_pkcs7_certificates(self, data):
+        utils._check_bytes("data", data)
+        bio = self._bytes_to_bio(data)
+        p7 = self._lib.PEM_read_bio_PKCS7(
+            bio.bio, self._ffi.NULL, self._ffi.NULL, self._ffi.NULL
+        )
+        if p7 == self._ffi.NULL:
+            raise ValueError("Unable to parse PKCS7 data")
+
+        p7 = self._ffi.gc(p7, self._lib.PKCS7_free)
+        return self._load_pkcs7_certificates(p7)
+
+    def load_der_pkcs7_certificates(self, data):
+        utils._check_bytes("data", data)
+        bio = self._bytes_to_bio(data)
+        p7 = self._lib.d2i_PKCS7_bio(bio.bio, self._ffi.NULL)
+        if p7 == self._ffi.NULL:
+            raise ValueError("Unable to parse PKCS7 data")
+
+        p7 = self._ffi.gc(p7, self._lib.PKCS7_free)
+        return self._load_pkcs7_certificates(p7)
+
+    def _load_pkcs7_certificates(self, p7):
+        nid = self._lib.OBJ_obj2nid(p7.type)
+        self.openssl_assert(nid != self._lib.NID_undef)
+        if nid != self._lib.NID_pkcs7_signed:
+            raise UnsupportedAlgorithm(
```

Comment:
Hmm, I feel like this is `ValueError`, not `UnsupportedAlgorithm`?

Reply:
You can have certificates inside a signedAndEnveloped structure I think, we just aren't going to support that at this time. That feels like `UnsupportedAlgorithm` to me, but if you feel strongly I'll change it.

Reply:
Ok, I'm perauded. Please attach the unsupported serialization reason flag to this though.